lib/Raven/Client.php

```diff
@@ -231,76 +110,67 @@ class Client
     /**
      * @var bool
      */
-    protected $_shutdown_function_has_been_set;
+    protected $_shutdown_function_has_been_set = false;
+
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
> how much code he was able to delete

This means nothing. I don\'t see any reason not to add a few lines of code to `Client::__construct` with calling `ClientBuilder::create([...])->getClient();`

```php
if (is_array($option)) {
    return ClientBuilder::create($option)->getClient();
}
```